src/elm/comparison.coffee

```diff
@@ -16,9 +16,9 @@ module.exports.Less = class Less extends Expression
       args[0].lessThan args[1]
     catch error
       if error instanceof IncompatibleTypesException
-        return null
-      else
         throw error
+      else
+        return null
```

Comment:
This flip of what happens doesn't make sense to me. Before it was catching the IncompatibleTypeException and returning a null, if the error was something else then it was re-throwing the error. After this change it eats and returns null for ANY error except IncompatibleTypeException. This doesn't seem correct to me.

Reply:
@mayerm94 can you field this question?  I don't mind updating if it needs to be updated.

Reply:
Based on the spec, we shouldn't be returning null if there is an incompatible type exception.  I believe @mayerm94 discussed with Moesel what should happen if a different type of error is generated.

Reply:
Relevant CQL spec section:
![image](https://user-images.githubusercontent.com/14879344/39884711-bbc6b8e2-5458-11e8-9fcd-06f5f990f7f3.png)

It does seem weird to me as well to return `null` for other types of errors rather than throwing.

Reply:
I can't imagine what other types of errors would pop up -- but if you all want to throw no matter what the error then feel free.  I think we could make an argument for either way.

Reply:
And yeah, I think I remember a discussion with @mayerm94 and I probably said returning null on other errors was fine.  I think there are other parts of CQL where unexpected things _do_ result in null.  Like I said, we could probably argue either side.

Reply:
@mayerm94 says he reviewed w moesel. that this is the expected behavior.

Reply:
Yeah, so to repeat -- @mayerm94 did review w/ me, and he did as we discussed.  TBH, I hadn't thought of the possibility that this might also eat an invalid UCUM unit error.  I'm totally fine with just always throwing, and in hindsight, that might be the better approach.  Sorry for leading you astray, @mayerm94.

Reply:
Sounds good.  I will update these accordingly.